nexus/src/app/instance.rs

```diff
@@ -1714,29 +1722,36 @@ impl super::Nexus {
             match vmm.runtime.state {
                 DbVmmState::Running
                 | DbVmmState::Rebooting
-                | DbVmmState::Migrating => {
-                    Ok((vmm.clone(), SocketAddr::new(vmm.propolis_ip.ip(), vmm.propolis_port.into())))
-                }
+                | DbVmmState::Migrating => Ok((
+                    vmm.clone(),
+                    SocketAddr::new(
+                        vmm.propolis_ip.ip(),
+                        vmm.propolis_port.into(),
+                    ),
+                )),
 
                 DbVmmState::Starting
                 | DbVmmState::Stopping
                 | DbVmmState::Stopped
                 | DbVmmState::Failed
-                | DbVmmState::Creating => {
+                | DbVmmState::Creating
+                | DbVmmState::Destroyed
+                | DbVmmState::SagaUnwound => {
                     Err(Error::invalid_request(format!(
-                        "cannot connect to serial console of instance in state \"{}\"",
```

Comment:
i tried to make this a stronger error type than `Error` here, so we could have nice "serial console" errors for the serial console path, and "propolis" errors for the `region{, snapshot}_replacement` saga path. but this runs into a bunch of new wrinkles like, `instance_lookup.lookup_for()` can fail in an `Error`-y way, `db_datastore.instance_fetch_with_vmm()` can fail in an `Error`-y way.

so after spinning a little too long on error plumbing here, i decided to take the different approach of picking the lowest common denominator message that makes sense for any caller of `propolis_addr_for_instance`. maybe "cannot connect to admin socket" or something would be more accurate, but i don't want to be confusing in a user-facing error message - there's no secret backdoor socket in your instance! we're just talking to the VMM!

Reply:
yeah, I think just making it generic is fine. 's fair. a slightly different wording is something like "this operation cannot be performed on instances in state ...", which is also generic but suggests that the specific thing you tried to do can't be done in that state. I worry a little that "administer an instance" is broad enough that it *does* include operations you can perform in these states --- i.e., is deleting an instance "administering" it?